### PR TITLE
docs: fix invalid link references & style the doc

### PR DESCRIPTION
**Pull request by Abinashbunty**
- Removed redundant link from the Code of Conduct header
- Referenced the correct discord `#help-and-questions` channel link.
- Added note for important info.
- Fixed grammar

## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Improve the layout and content of the file.

## Test Plan

Test plan not applicable for this PR. 😄

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -7,25 +7,25 @@ The [Open Source Guides](https://opensource.guide/) website has a collection of
...
 - Take a look at the [features requested](https://github.com/facebook/docusaurus/labels/feature) by others in the community and consider opening a pull request if you see something you want to work on.
 
 Contributions are very welcome. If you think you need help planning your contribution, please ping us on Twitter at [@docusaurus](https://twitter.com/docusaurus) and let us know you are looking for a bit of help.
 
 ### Join our Discord Channel
 
-We have the [`#contributors`](https://discord.gg/6g6ASPA) channel on [Discord](https://discord.gg/docusaurus) to discuss all things about Docusaurus development. You can also be of great help by helping other users in the help channel.
+We have the [`#contributors`](https://discord.gg/6g6ASPA) channel on [Discord](https://discord.gg/docusaurus) to discuss all things about Docusaurus development. You can also be of great help by helping other users in the [`#help-and-questions`](https://discord.gg/GjwjTDasTF) channel.
```
Comment (Josh-Cena) [inline CONTRIBUTING.md:28]:
```suggestion
We have the [`#contributors`](https://discord.gg/6g6ASPA) channel on [Discord](https://discord.gg/docusaurus) to discuss all things about Docusaurus development. You can also be of great help by helping other users in the [`#help-and-questions`](https://discord.gg/fwbcrQ3dHR) channel.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ Docusaurus has one primary branch `main` and we use feature branches with deploy
 
 When [opening a new issue](https://github.com/facebook/docusaurus/issues/new/choose), always make sure to fill out the issue template. **This step is very important!** Not doing so may result in your issue not being managed in a timely fashion. Don't take this personally if this happens, and feel free to open a new issue once you've gathered all the information required by the template.
 
-**Please don't use the GitHub issue tracker for questions.** If you have questions about using Docusaurus, use any of our [support channels](https://docusaurus.io/community/support), and we will do our best to answer your questions.
+> **Note**: **Please don't use the GitHub issue tracker for questions.** If you have questions about using Docusaurus, use any of our [support channels](https://docusaurus.io/community/support), and we will do our best to answer your questions.
```
Comment (Josh-Cena) [inline CONTRIBUTING.md:53]: We don't use `> **Note:**`. We render it on our website, where the notecard syntax is not supported. It's also not good practice (and I regret implementations that use this syntax) to overload the blockquote semantics.

Comment (Abinashbunty) [reply CONTRIBUTING.md:53]: I will change this blockquote content to back how it was then. 👍🏻

**Review by Josh-Cena — APPROVED** [review]
Thank you! Looks great.

**Pull request merged**